api/v1/checks.go

```diff
@@ -139,6 +139,60 @@ func (c S3BucketCheck) GetType() string {
 	return "s3Bucket"
 }
 
+type GCPConnection struct {
+	Endpoint        string          `yaml:"endpoint" json:"endpoint,omitempty"`
+	CredentialsJSON *kommons.EnvVar `yaml:"credentialsJSON" json:"credentialsJSON,omitempty"`
+}
+
+// GCSQuery represents a query to filter objects from a bucket.
+type GCSQuery struct {
+	// Delimiter returns results in a directory-like fashion.
+	// Results will contain only objects whose names, aside from the
+	// prefix, do not contain delimiter. Objects whose names,
+	// aside from the prefix, contain delimiter will have their name,
+	// truncated after the delimiter, returned in prefixes.
+	// Duplicate prefixes are omitted.
+	// Optional.
+	Delimiter string `yaml:"delimiter,omitempty" json:"delimiter,omitempty"`
+
+	// Prefix is the prefix filter to query objects
+	// whose names begin with this prefix.
+	// Optional.
+	Prefix string `yaml:"prefix,omitempty"  json:"prefix,omitempty"`
+
+	// Versions indicates whether multiple versions of the same
+	// object will be included in the results.
+	Versions bool `yaml:"versions,omitempty" json:"versions,omitempty"`
+
+	// StartOffset is used to filter results to objects whose names are
+	// lexicographically equal to or after startOffset. If endOffset is also set,
+	// the objects listed will have names between startOffset (inclusive) and
+	// endOffset (exclusive).
+	StartOffset string `yaml:"startOffset,omitempty" json:"startOffset,omitempty"`
+
+	// EndOffset is used to filter results to objects whose names are
+	// lexicographically before endOffset. If startOffset is also set, the objects
+	// listed will have names between startOffset (inclusive) and endOffset (exclusive).
+	EndOffset string `yaml:"endOffset,omitempty" json:"endOffset,omitempty"`
+}
+
+type GCSBucketCheck struct {
+	Description   `yaml:",inline" json:",inline"`
+	Templatable   `yaml:",inline" json:",inline"`
+	FolderTest    `yaml:",inline" json:",inline"`
+	GCPConnection `yaml:",inline" json:",inline"`
+	*GCSQuery     `yaml:",inline" json:",inline"`
```

Comment:
I think we only want a path (a regex for matching the filename)

The prefix should be derived from the bucket (id specified)